subsys/net/l2/Kconfig

```diff
@@ -69,6 +69,8 @@ endif # NET_L2_WIFI_MGMT
 
 config NET_L2_WIFI_SHELL
 	bool "Wi-Fi shell module"
+	depends on SHELL
```

Comment:
What about depending here to NET_SHELL so user can disable network shell support by that symbol.
Same comment to other net related depends